playerbot/strategy/actions/UseRingAction.cpp

```diff
@@ -0,0 +1,36 @@
+#include "botpch.h"
+#include "../../playerbot.h"
+#include "UseRingAction.h"
+#include "Item.h"
+#include "ItemPrototype.h"
+#include "Player.h"
+
+using namespace ai;
+
+bool UseRingAction::Execute(Event& event)
+{
+	auto rings = AI_VALUE(list<Item*>, "rings on use");
+
+	if (rings.empty())
+		return false;
+
+	for (Item * item : rings)
+	{
+		ItemPrototype const* proto = item->GetProto();
```

Comment:
`const ItemPrototype* proto = item->GetProto();`

Reply:
In server files ItemPrototype const* proto is used

Reply:
cmangos code doesn't follow any standards, but I'm trying to improve that atleast in our side